### PR TITLE
Update eslint and eslint-config-cesium

### PR DESCRIPTION
**Pull request by ggetz**
# Description

Following up from https://github.com/CesiumGS/eslint-config-cesium/pull/6, this bumps the version of eslint and corresponding packages. The major change is a migration from `.eslintrc.js` files to [flat config files](https://eslint.org/docs/latest/use/configure/configuration-files).

Other notable changes include:
 - Added a `package.json` and corresponding config to `.github/actions/check-for-cla`. This was mainly to make eslint happy, but is probably better practice anyway to ensure version compatibility with dependencies.
 - Removed outdated inline eslint comments where found
- Bumped minimum supported version for compatibility, which matches the current LTS of NodeJS anyway

## Issue number and link

N/A

## Testing plan

1. Run `npm eslint` and confirm there are no issues
1. Introduce an eslint error to ensure that errors still surface when running `npm eslint`.

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [-] ~I have updated `CHANGES.md` with a short summary of my change~
- [-] ~I have added or updated unit tests to ensure consistent code coverage~
- [-] ~I have update the inline documentation, and included code examples where relevant~
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by jjspace** [comment]
@ggetz `make-zip` is failing for a missing `.eslintignore` file

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        { vars: "all", args: "none", caughtErrors: "none" },
+      ],
+    },
+  },
+  {
+    files: ["Apps/Sandcastle/**/*", "Apps/TimelineDemo/**/*"],
+    languageOptions: {
+      sourceType: "script",
+      globals: {
+        ...globals.amd,
```
[the remark this answers is not in the document]
Comment (ggetz) [reply eslint.config.js:59]: Strictly speaking, the way we had it configured before, `"requre"` was redundant. 

`globals.amd` includes `require` and `defined`, which are used in both TimelineDemo and Sandcastle.

File: eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    },
+  },
+  {
+    files: ["Apps/Sandcastle/load-cesium-es6.js"],
+    languageOptions: {
+      sourceType: "module",
+    },
+  },
+  {
+    files: ["Specs/**/*", "packages/**/Specs/**/*"],
```
[the remark this answers is not in the document]
Comment (ggetz) [reply eslint.config.js:78]: They are now covered in the `**/**.cjs` blob.

**Comment by ggetz** [comment]
@jjspace This should be ready for another look.

**Comment by jjspace** [comment]
Apparently prettier didn't like that merge but I touched it up quick.

LGTM, thanks @ggetz!

**Pull request merged**